### PR TITLE
VIA Support for the Idobo (ID75)

### PR DESCRIPTION
**Pull request by ifohancroft**
I have added VIA support for the Idobo (ID75) keyboard.

### CODE REVIEW

File: keyboards/idobo/keymaps/via/rules.mk [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BOOTMAGIC_ENABLE = light
```
Comment (fauxpark) [inline keyboards/idobo/keymaps/via/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite
```

File: keyboards/idobo/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+ [0] = LAYOUT_ortho_5x15(
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV,  \
```
Comment (fauxpark) [inline keyboards/idobo/keymaps/via/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/idobo/config.h [hunk 3]
```diff
@@ -20,11 +20,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
+#define VENDOR_ID       0x1080 // Idobo - 1d080 - 1080
 #define PRODUCT_ID      0x0000
 #define DEVICE_VER      0x0001
 #define MANUFACTURER    You
```
Comment (fauxpark) [inline keyboards/idobo/config.h]:
```suggestion
#define MANUFACTURER    Massdrop
```

Apparently this is also known as the "MD75"?
https://drop.com/buy/xd75-aluminum-mechanical-keyboard (from the readme)

Comment (noroadsleft) [reply keyboards/idobo/config.h]: Massdrop doesn't market this as a Massdrop-produced board. As near as I can tell the actual manufacturer is unknown.

I did also find a purchase listing for it [on AliExpress](https://www.aliexpress.com/i/32952542823.html).

Comment (ifohancroft) [reply keyboards/idobo/config.h]: This is the https://drop.com/buy/id75-hot-swappable-ortholinear-keyboard-kit
I don't think it's the same as the XD75

File: keyboards/idobo/keymaps/via/keymap.c (hunk 2 above)
Comment (ifohancroft) [reply keyboards/idobo/keymaps/via/keymap.c]: Thanks! I didn't know that. I agree that it looks neater without them. I will fix it in a bit.

File: keyboards/idobo/keymaps/via/rules.mk (hunk 1 above)
Comment (ifohancroft) [reply keyboards/idobo/keymaps/via/rules.mk]: I can't believe I made that mistake :D
Thanks! I'll fix it in a bit.

File: keyboards/idobo/config.h (hunk 3 above)
Comment (fauxpark) [reply keyboards/idobo/config.h]: That's... odd, I could have sworn the link in the readme (and the link I pasted) was this: https://drop.com/buy/75keys-aluminum-mechanical-keyboard-kit

Comment (ifohancroft) [reply keyboards/idobo/config.h]: > That's... odd, I could have sworn the link in the readme (and the link I pasted) was this: https://drop.com/buy/75keys-aluminum-mechanical-keyboard-kit

Hmm, that's odd indeed. The link you just posted seems to be the same as the XD75.

Yes, I remember seeing a file in the board's folder pointing to XD75 and I've been meaning to send a pull request about it but I forgot.

Comment (fauxpark) [reply keyboards/idobo/config.h]: From the comments:
> If you are having issues finding the correct QMK keyboard this is, you can find it as "idobo" in QMK

Comment (ifohancroft) [reply keyboards/idobo/config.h]: Just fixed the readme

File: keyboards/idobo/config.h [hunk 4]
```diff
@@ -20,11 +20,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
+#define VENDOR_ID       0x8010
```
Comment (fauxpark) [inline keyboards/idobo/config.h]:
```suggestion
#define VENDOR_ID       0x6964 // ID
```

How about these?

File: keyboards/idobo/config.h [hunk 5]
```diff
@@ -20,11 +20,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 #define PRODUCT_ID      0x0000
```
Comment (fauxpark) [inline keyboards/idobo/config.h]:
```suggestion
#define PRODUCT_ID      0x0075
```

File: keyboards/idobo/config.h (hunk 3 above)
Comment (fauxpark) [reply keyboards/idobo/config.h]: In any case, the vendor name should be something other than "You", that's from the templates I think. Maybe "Generic"?

File: keyboards/idobo/config.h (hunk 5 above)
Comment (ifohancroft) [reply keyboards/idobo/config.h]: Good call.

File: keyboards/idobo/config.h (hunk 3 above)
Comment (ifohancroft) [reply keyboards/idobo/config.h]: Fixed. Thanks for the fast reviews and sry for the multiple commits/not fixing some things in the first place.

File: keyboards/idobo/config.h (hunk 4 above)
Comment (ifohancroft) [reply keyboards/idobo/config.h]: Any particular reason for this?

Comment (fauxpark) [reply keyboards/idobo/config.h]: It seems to be the general convention for VIA support:
https://caniusevia.com/docs/configuring_qmk/#change-vendor_id-product_id

Comment (ifohancroft) [reply keyboards/idobo/config.h]: I can't figure out what am I missing. Isn't 0x8010 a valid Vendor ID value?

**Comment by noroadsleft** [comment]
Thanks!

**Comment by ifohancroft** [comment]
> Thanks!

You too!

**Pull request merged**